corehq/ex-submodules/casexml/apps/phone/caselogic.py

```diff
@@ -597,23 +627,27 @@ def filter_cases_modified_elsewhere_since_sync(cases, last_sync):
         # { case_id: [{'token': 'token value', 'date': 'date value'}, ...]}
         all_case_updates_by_sync_token = defaultdict(list)
         for row in modification_dates:
-            # incoming format is a list of objects that look like this:
+            # format from couch is a list of objects that look like this:
             # {
             #   'value': '2012-08-22T08:55:14Z', (most recent date updated)
-            #   'key': ['[case id]', '[sync token id]']
+            #   'key': ['case-id', 'sync-token-id']
             # }
             if row['value']:
-                all_case_updates_by_sync_token[row['key'][0]].append(
-                    {'token': row['key'][1], 'date': datetime.strptime(row['value'], '%Y-%m-%dT%H:%M:%SZ')}
-                )
+                modification_date = datetime.strptime(row['value'], '%Y-%m-%dT%H:%M:%SZ')
+                if modification_date >= last_sync_token.date:
+                    case_id, sync_token_id = row['key']
+                    all_case_updates_by_sync_token[case_id].append(
+                        {'token': sync_token_id, 'date': modification_date}
+                    )
```

Comment:
don't bother adding case updates that are before the last sync to the data structure